apps/src/applab/firebaseCounters.js

```diff
@@ -110,42 +100,10 @@ export function incrementRateLimitCounters() {
         }
       }));
     });
-    return Promise.all(promises);
-  });
-}
-
-/**
- * Returns a promise which resolves once Firebase config has been fetched from
- * the server and cached locally. Also starts listening for future changes to config.
- * @returns {Promise<>}
- */
-function loadConfig() {
-  if (tableRowCountLimit && rateLimitMap) {
-    // The firebase config has already been loaded.
-    return Promise.resolve();
-  }
-
-  const configRef = getConfig();
-  return configRef.once('value', snapshot => {
-    handleLoadConfig(snapshot.val());
-
-    // Make sure we don't listen multiple times.
-    configRef.off();
-
-    // Update globals to reflect firebase config any time it changes in the future.
-    configRef.on('value', snapshot => handleLoadConfig(snapshot.val()));
+    return Promise.all(intervalPromises);
```

Comment:
You've nearly reduced this to a `map` operation.  Could:

```javascript
return Promise.all(Object.keys(config.limits).map(interval => {
  const limitRef = getDatabase(Applab.channelId).child(`counters/limits/${interval}`);
  return limitRef.transaction(/* ... */);
}));
```

Up to you whether that would read better.